browser/lib/transport/jsonptransport.js

```diff
@@ -17,8 +18,10 @@ var JSONPTransport = (function() {
 	}
 	Utils.inherits(JSONPTransport, CometTransport);
 
-	JSONPTransport.isAvailable = function() { return true; };
-	ConnectionManager.httpTransports['jsonp'] = ConnectionManager.transports['jsonp'] = JSONPTransport;
+	JSONPTransport.isAvailable = function() { return isSupported; };
+	if(isSupported) {
```

Comment:
I disagree, I think what's been done is correct. All other transports only add themselves to the `transports` array if they're supported (and connectionManager occasionally assumes that). If someone using React uses `transports: ['jsonp']` (though I don't imagine anyone would), they'll get a "*no requested transports available*" error, same as if someone using IE8 specified `transports: ['web_socket']`.